### PR TITLE
S004 t39 paging filtering orders

### PR DESCRIPTION
**Pull request by pkr5**

### CODE REVIEW

File: src/main/java/com/project/stockexchangeappbackend/repository/OrderRepository.java [hunk 1]
```diff
@@ -17,4 +23,12 @@
     @DBQueryMeasureTime
     Optional<Order> findById(Long id);
 
+    @Override
+    @DBQueryMeasureTime
+    @Query(value = "SELECT * FROM ALL_ORDERS",
+            countQuery = "SELECT COUNT(*) FROM ALL_ORDERS",
+            nativeQuery = true)
+    Page<Order> findAll(@Nullable Specification specification, Pageable pageable);
```
Comment (karolcymerys) [inline src/main/java/com/project/stockexchangeappbackend/repository/OrderRepository.java]: Isn't is useless now?

File: src/main/java/com/project/stockexchangeappbackend/repository/specification/OrderSpecification.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        @Spec(path = "amount", spec = LessThanOrEqual.class, params = {"amount<"}),
+        @Spec(path = "amount", spec = Equal.class, params = {"amount"}),
+        @Spec(path = "price", spec = GreaterThanOrEqual.class, params = {"price>"}),
+        @Spec(path = "price", spec = LessThanOrEqual.class, params = {"price<"}),
+        @Spec(path = "price", spec = Equal.class, params = {"price"}),
+        @Spec(path = "s.name", spec = LikeIgnoreCase.class, params = {"name"}),
+        @Spec(path = "s.abbreviation", spec = LikeIgnoreCase.class, params = {"abbreviation"}),
+})
+
+public interface OrderSpecification extends Specification<Order> {
```
Comment (karolcymerys) [inline src/main/java/com/project/stockexchangeappbackend/repository/specification/OrderSpecification.java]: Isn't is useless now?

File: src/main/java/com/project/stockexchangeappbackend/rest/OrderController.java [hunk 3]
```diff
@@ -42,8 +45,16 @@ public OrderDTO getOrderDetails(@ApiParam(value = "Id of a order.", required = t
     @ApiOperation(value = "Create new order")
     @ApiResponses(@ApiResponse(code = 200, message = "Order was successfully created."))
     public void createOrder(@ApiParam(value = "Order object to create.", required = true)
-                                @RequestBody @Valid OrderDTO orderDTO) {
+                            @RequestBody @Valid OrderDTO orderDTO) {
         orderService.createOrder(orderDTO);
     }
 
+    @GetMapping
+    //@PreAuthorize("hasAnyRole('ADMIN','USER')")
```
Comment (karolcymerys) [inline src/main/java/com/project/stockexchangeappbackend/rest/OrderController.java]: ?

**Review by karolcymerys — CHANGES_REQUESTED** [review]

File: src/main/java/com/project/stockexchangeappbackend/rest/OrderController.java (hunk 3 above)
Comment (pkr5) [reply src/main/java/com/project/stockexchangeappbackend/rest/OrderController.java]: my bad

File: src/main/java/com/project/stockexchangeappbackend/service/OrderServiceImpl.java [hunk 4]
```diff
@@ -57,6 +59,14 @@ public void createOrder(OrderDTO orderDTO) {
         orderRepository.save(order);
     }
 
+    @Override
+    @LogicBusinessMeasureTime
+    public Page<AllOrders> findAllOrders(Pageable pageable, Specification<AllOrders> specification) {
+        Page<AllOrders> page = allOrdersRepository.findAll(specification, pageable);
+        return page;
+        //return allOrdersRepository.findAll(specification, pageable);
```
Comment (karolcymerys) [inline src/main/java/com/project/stockexchangeappbackend/service/OrderServiceImpl.java]: remove comment and refactor it

**Review by karolcymerys — APPROVED** [review]

**Pull request merged**